jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -76,6 +75,11 @@
      */
     private SafeArrayList<Mesh> targets = new SafeArrayList<Mesh>(Mesh.class);
     /**
+     * Geometry with an animated mesh, for calculating attachments node
+     * transforms. A null means no geometry is subject to this control.
+     */
+    private Geometry targetGeometry = null;
```

Comment:
I meant the targetGeomerty shouldn't be in SkeletonControl. It should only be in Bone and the target computed in Bone.
Even with your implementation it's just a matter of picking the last entry of the targets list, we don't need an additional variable.